Develop3

Anayalab opened the pull request.

Commit:
feat(tests): Добавлены тесты для Stellar Burgers

Commit:
feat(DriverFactory): добавлен headless режим через системное свойство

Commit:
Добавлены результаты тестов в allure-results

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/stellarburgers/model/ForgotPasswordPage.java:
```diff
@@ -0,0 +1,28 @@
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+public class ForgotPasswordPage extends BasePage {
```
Отлично что есть иерархия po

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/stellarburgers/model/MainPage.java:
```diff
@@ -0,0 +1,122 @@
+    public boolean isBunsTabActive() {
+        // Просто проверяем, что раздел "Булки" виден
+        try {
+            wait.until(ExpectedConditions.visibilityOf(bunsTab));
```
Нужно смотреть на класс, точнее его кусочек, который появляется именно при скролле. Перепроверь, пожалуйста

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/stellarburgers/tests/NavigationTest.java:
```diff
@@ -0,0 +1,71 @@
+        String currentUrl = driver.getCurrentUrl();
+        System.out.println("URL after navigation to constructor by logo: " + currentUrl);
+        Assert.assertTrue("Пользователь должен быть перенаправлен на главную страницу. URL: " + currentUrl, 
+                currentUrl.equals("https://stellarburgers.nomoreparties.site/") || 
```
Все урлы выносим как константы 
Нужно исправить

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/stellarburgers/tests/RegistrationTest.java:
```diff
@@ -0,0 +1,85 @@
+        registerPage.clickRegisterButton();
+        
+        try {
+            Thread.sleep(2000);
```
Все слипы заменяем на явные ожидания 
Нужно исправить

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/stellarburgers/utils/BaseTest.java:
```diff
@@ -0,0 +1,29 @@
+
+    @Before
+    public void setUp() {
+        String browser = System.getProperty("browser", "chrome");
```
Отлично сделан выбор браузера

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/stellarburgers/utils/DriverFactory.java:
```diff
@@ -0,0 +1,144 @@
+                    return path;
+                }
+            }
+        } else if (os.contains("linux")) {
```
Это очень правильно)

pomegranate282922 reviewed (commented):
Все классно, кроме нескольких замечаний:
1. Поправить проверку в тестах на переходы 
2. Убрать слипы 
3. Вынести урлы

Commit:
feat(MainPage): изменен локатор вкладки "Булки" на xpath с проверкой по классу

Commit:
feat(utils): Добавлен класс Constants для хранения URL

Перенесены BASE_URL и API_BASE_URL в отдельный класс Constants для централизованного управления константами. Обновлены ссылки в BaseTest, UserApiClient и тестах.

Commit:
fix(tests): заменены явные ожидания через Thread.sleep на WebDriverWait в LogoutTest и RegistrationTest

pomegranate282922 commented on src/test/java/ru/yandex/praktikum/stellarburgers/model/MainPage.java:
```diff
@@ -20,7 +20,7 @@ public class MainPage extends BasePage {
     private WebElement constructorButton;
 
-    @FindBy(css = "span.text_type_main-default")
+    @FindBy(xpath = "//span[contains(@class, 'text_type_main-default') and contains(text(), 'Булки')]")
```
Вообще имелось в виду анализировать наличие слова current в классах элемента. 
можно улучшить

pomegranate282922 reviewed (approved):
Ок, зачет!
